src/push_context.rs

```diff
@@ -0,0 +1,397 @@
+
+        let rolling_minor_with_postfix_or_tag = if cli.rolling_minor.0.is_some() || cli.rolling {
+            format!(
+                "{rolling_prefix_or_tag}.{}+rev-{}",
+                commit_count, git_ctx.revision_info.revision
+            )
+        } else {
+            rolling_prefix_or_tag.to_string() // This will always be the tag since `self.rolling_prefix` was empty.
+        };
+
+        // STEP: calculate labels
+        let merged_labels = {
+            let mut labels: HashSet<_> = cli
+                .extra_labels
+                .clone()
+                .into_iter()
+                .filter(|v| !v.is_empty())
+                .collect();
+            let extra_tags: HashSet<_> = cli
+                .extra_tags
+                .clone()
+                .into_iter()
+                .filter(|v| !v.is_empty())
+                .collect();
+
+            if !extra_tags.is_empty() {
+                let message = "`extra-tags` is deprecated and will be removed in the future. Please use `extra-labels` instead.";
+                tracing::warn!("{message}");
+
+                if matches!(&exec_env, ExecutionEnvironment::GitHub) {
+                    println!("::warning::{message}");
+                }
+
+                if labels.is_empty() {
+                    labels = extra_tags;
+                } else {
+                    let message =
+                        "Both `extra-tags` and `extra-labels` were set; `extra-tags` will be ignored.";
+                    tracing::warn!("{message}");
+
+                    if matches!(exec_env, ExecutionEnvironment::GitHub) {
+                        println!("::warning::{message}");
+                    }
+                }
+            }
+
+            // Get the "topic" labels from git_ctx, extend local mut labels
+            let topics = git_ctx.repo_topics;
+            labels = labels
+                .into_iter()
+                .chain(topics.iter().cloned())
+                .collect::<HashSet<String>>();
+
+            // Here we merge explicitly user-supplied labels and the labels ("topics")
+            // associated with the repo. Duplicates are excluded and all
+            // are converted to lower case.
+            let merged_labels: Vec<String> = labels
+                .into_iter()
+                .take(MAX_NUM_TOTAL_LABELS)
+                .map(|s| s.trim().to_lowercase())
+                .filter(|t: &String| {
+                    !t.is_empty()
+                        && t.len() <= MAX_LABEL_LENGTH
+                        && t.chars().all(|c| c.is_alphanumeric() || c == '-')
+                })
+                .collect();
+
+            merged_labels
+        };
+
+        // flake_dir is an absolute path of flake_root(aka git_root)/subdir
+        let flake_dir = local_git_root.join(&subdir);
+        // TODO(review): depending on what the user called us with, this flake_dir isn't even necessarily canonicalized, is this a sec/traversal issue?
```

Comment:
Make a ticket to check this?

Reply:
we talked about this, it's not an issue due to checks above, removed this comment